packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -108,6 +109,44 @@ function getVisibleLabels(
   );
 }
 
+function getDefaultTextAnchor(
+  angle: number,
+  position: 'top' | 'bottom' | 'none' | undefined,
+): ChartsTextStyle['textAnchor'] {
+  const adjustedAngle = clampAngle(angle);
+
+  if (adjustedAngle === 0 || adjustedAngle === 180) {
```

Comment:
The usage of `===` feels a bit weir when playing with it, but I assume nobody will use axes with something else than 0, 45, or 90

Reply:
> The usage of `===` feels a bit weir when playing with it, but I assume nobody will use axes with something else than 0, 45, or 90

Yeah, that's true. The jump from 0 to 1 is pretty noticeable, but I'm assuming people will use specific angles. 

We can make it smoother, but I didn't think it was worth the effort. If you think it would make sense, I can put some effort into it